src/compiler/factory/emitHelpers.ts

```diff
@@ -961,6 +967,29 @@ namespace ts {
             };`
     };
 
+    /**
+     * Parameters:
+     *  @param receiver — The object being checked if it has the private member.
+     *  @param state — One of the following:
+     *      - A WeakMap when the member is a private instance field.
+     *      - A WeakSet when the member is a private instance method or accessor.
+     *      - A function value that should be the undecorated class constructor when the member is a private static field, method, or accessor.
+     *
+     * Usage:
+     * This helper is used to transform `#field in expression` to
+     *      `__classPrivateFieldIn(expression, <weakMap/weakSet/constructor>)`
+     */
+    export const classPrivateFieldInHelper: UnscopedEmitHelper = {
+        name: "typescript:classPrivateFieldIn",
+        importName: "__classPrivateFieldIn",
+        scoped: false,
+        text: `
+            var __classPrivateFieldIn = (this && this.__classPrivateFieldIn) || function(receiver, state) {

```

Comment:
Would it make sense to have `receiver` and `state` swapped here? For the other `__classPrivateField` helpers, `receiver` comes before `state` because that's *essentially* how it evaluates left-to-right:

```js
// source
receiver.#state

// output
__classPrivateFieldGet(receiver, _state)
```

Maintaining this order is useful for source maps. 

However, the `in` expression swaps the order, so it would make more sense if they were also reversed to improve source mapping:

```js
// source
#state in receiver

// proposed output
__classPrivateFieldIn(__state, receiver)
```

Reply:
Ah yes, the transformed code also reads much nicer now too. I like it.